internal/cli/cli.go

```diff
@@ -58,5 +61,60 @@ func Run() {
 		fmt.Printf("payment.skey=%s\n", bursa.GetKeyFile(w.PaymentSKey))
 		fmt.Printf("stake.vkey=%s\n", bursa.GetKeyFile(w.StakeVKey))
 		fmt.Printf("stake.skey=%s\n", bursa.GetKeyFile(w.StakeSKey))
-	} // TODO: output to files
+	} else {
+		fmt.Printf("Output dir: %v\n", *flagOutput)
+		_, err := os.Stat(*flagOutput)
+		if os.IsNotExist(err) {
+			err = os.MkdirAll(*flagOutput, 0755)
+			if err != nil {
+				panic(err)
+			}
+		}
+		var fileMap = []map[string]string{
+			{"seed.txt": w.Mnemonic},
+			{"payment.addr": w.PaymentAddress},
+			{"stake.addr": w.StakeAddress},
+			{"payment.vkey": bursa.GetKeyFile(w.PaymentVKey)},
+			{"payment.skey": bursa.GetKeyFile(w.PaymentSKey)},
+			{"stake.vkey": bursa.GetKeyFile(w.StakeVKey)},
+			{"stake.skey": bursa.GetKeyFile(w.StakeSKey)},
+		}
+		var g errgroup.Group
+		for _, m := range fileMap {
+			for k, v := range m {
+				k := k
+				v := v
+				g.Go(func() error {
+					path := filepath.Join(*flagOutput, k)
+					_, err := os.Stat(path)
+					if os.IsNotExist(err) {
+						file, err := os.Create(path)
+						if err == nil {
+							defer file.Close()
+							_, err := file.Write([]byte(v))
+							if err != nil {
+								return err
+							}
+						}
+						return err
+					}
```

Comment:
Creating the output directory inside multiple concurrent goroutines is a bad idea. This should be done once, outside of this loop

Reply:
It's not done inside a loop.